src/SynthImage/line_extraction.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+from PIL import Image, ImageDraw, ImageFont
+
+
+class ExtractLines:
+    def __init__(self, aug_img, page_text, rotation_angle, font_size, font_path):
+        self.aug_img = aug_img
+        self.page_text = page_text
+        self.rotation_angle = rotation_angle
+        self.font_size = font_size
+        self.font_path = font_path
+
+    def extract_lines(self):
+        if self.rotation_angle != 0:
+            self.aug_img = self.aug_img.rotate(
+                -self.rotation_angle, expand=True, fillcolor=(255, 255, 255)
+            )
+
+        lines = self.page_text.split("\n")
+        font = ImageFont.truetype(self.font_path, self.font_size, encoding="utf-16")
+        line_images = []
+
+        # Create a blank image to draw text
+        blank_img = Image.new("RGB", self.aug_img.size, (255, 255, 255))
+
+        # Step 1: Determine the maximum width
```

Comment:
please modularise the step one in a method. U can do that by defining a method which will return u the max width. Do so u can test the main method in more efficient.